internal/processor.go

```diff
@@ -153,10 +154,20 @@ func (p *MessageProcessor) callback(ctx context.Context, payload []byte, msg *na
 			return fmt.Errorf("no schema found for for: %s %v for msgid: %s", modelVersionId, foundSchema.Data, msgid)
 		}
 	}
-	if err := p.provider.Process(data, schema); err != nil {
-		p.logger.Error("error processing change event: %s. %s", data, err)
-		return err
+	var wg sync.WaitGroup
+
+	for _, provider := range p.providers {
```

Comment:
may not matter at this point but spinning up go routines like this would be a better use case for a worker pattern where you have N number of worker go routines and they all pull from a channel.

something like (psuedo code)

```
for i = 0; i < 5; c++ {
  go startProcessor()
}

func startProcessor {
  for data := range chan {
    process(data)
  }
}

```